eng/build.ps1

```diff
@@ -100,6 +100,34 @@ function Get-Help() {
   Write-Host "For more information, check out https://github.com/dotnet/runtime/blob/master/docs/workflow/README.md"
 }
 
+function Assert-InstalledDependency($dependencyName)
+{
+  Write-Host "Checking for $dependencyName..."
+  if (((Get-ItemProperty HKLM:\Software\Microsoft\Windows\CurrentVersion\Uninstall\*).DisplayName -Match
```

Comment:
I have all these dependencies (except Visual Studio) installed via xcopy on my machine. I do not have any of this in registry. I know number of other folks on the team are using similar setup. CI is using this setup too.

This should check whether the dependency is on the path.